### PR TITLE
[MC,AArch64] Create mapping symbols with non-unique names

### PR DESCRIPTION
**Pull request by MaskRay**
Add `createLocalSymbol` to create a local, non-temporary symbol.
Different from `createRenamableSymbol`, the `Used` bit is ignored,
therefore multiple local symbols might share the same name.

Utilizing `createLocalSymbol` in AArch64 allows for efficient mapping
symbol creation with non-unique names, saving .strtab space.
The behavior matches GNU assembler.

### CODE REVIEW

**Review by aengelke — APPROVED** [review]
LGTM, this now matches the behavior of GNU as.

File: llvm/lib/Target/AArch64/MCTargetDesc/AArch64ELFStreamer.cpp [hunk 1]
```diff
@@ -271,8 +271,7 @@ class AArch64ELFStreamer : public MCELFStreamer {
   }
 
   void emitMappingSymbol(StringRef Name) {
-    auto *Symbol = cast<MCSymbolELF>(getContext().getOrCreateSymbol(
-        Name + "." + Twine(MappingSymbolCounter++)));
+    auto *Symbol = cast<MCSymbolELF>(getContext().createLocalSymbol(Name));
```
Comment (smithp35) [inline llvm/lib/Target/AArch64/MCTargetDesc/AArch64ELFStreamer.cpp:273]: I think the removed code is the only use for MappingSymbolCounter, can it be removed?

**Review by smithp35 — COMMENTED** [review]
I think MappingSymbolCounter can be removed now, otherwise LGTM.

File: llvm/lib/Target/AArch64/MCTargetDesc/AArch64ELFStreamer.cpp (hunk 1 above)
Comment (MaskRay) [reply llvm/lib/Target/AArch64/MCTargetDesc/AArch64ELFStreamer.cpp:273]: Removed by the second commit. Thanks

**Pull request merged**